src/ClientData/CaptureData.cpp

```diff
@@ -144,6 +144,12 @@ void CaptureData::ComputeVirtualAddressOfInstrumentedFunctionsIfNecessary(
         module_data->ConvertFromOffsetInFileToVirtualAddress(instrumented_function.file_offset());
     instrumented_function.set_function_virtual_address(virtual_address);
     ++updated_function_count;
```

Comment:
I think, the increment should still be at the end of the loop. Like we've undated the data in the CaptureData, in the ScopeIdProvider and finally do the increment.